src/Console/Command/CompareCommand.php

```diff
@@ -0,0 +1,117 @@
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Console\Command;
+
+use PhpCsFixer\Config;
+use PhpCsFixer\ConfigInterface;
+use PhpCsFixer\Console\ConfigurationResolver;
+use PhpCsFixer\FixerFactory;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Helper\Table;
+use Symfony\Component\Console\Helper\TableCell;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+
+/**
+ * @author Adamo Aerendir Crespi <hello@aerendir.me>
+ */
+final class CompareCommand extends Command
+{
+    const COMMAND_NAME = 'compare';
+
+    /**
+     * @var ConfigInterface
+     */
+    private $defaultConfig;
+
+    /**
+     * @var FixerFactory
+     */
+    private $fixerFactory;
+
+    /**
+     * @param null|FixerFactory $fixerFactory
+     */
+    public function __construct(FixerFactory $fixerFactory = null)
+    {
+        parent::__construct();
+
+        if (null === $fixerFactory) {
+            $fixerFactory = new FixerFactory();
+            $fixerFactory->registerBuiltInFixers();
+        }
+
+        $this->fixerFactory = $fixerFactory;
+        $this->defaultConfig = new Config();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure()
+    {
+        $this
+            ->setName(self::COMMAND_NAME)
+            ->setDefinition(
+                [
+                    new InputOption('config', '', InputOption::VALUE_REQUIRED, 'The path to a .php_cs file.'),
+                ]
+            )
+            ->setDescription('Compares existent features with the ones actually configured.')
+        ;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $passedConfig = $input->getOption('config');
+
+        $resolver = new ConfigurationResolver(
+            $this->defaultConfig,
+            [
+                'allow-risky' => true,
+                'config' => $passedConfig,
+            ],
+            getcwd()
+        );
+
+        $configured = $resolver->getFixers();
+        $builtIn = $this->fixerFactory->getFixers();
+
+        usort($builtIn, function ($a, $b) {
+            return strcmp($a->getName(), $b->getName());
+        });
+
+        $table = new Table($output);
+        $table->setHeaders([
+            [new TableCell(sprintf('Found <fg=yellow;>%s built-in</> fixers. Of those, <fg=yellow;>%s are configured</> to actually be used.', count($builtIn), count($configured)), ['colspan' => 3])],
+            ['Fixer', 'In use', 'Is risky'],
+        ]);
+
+        $body = [];
+        foreach ($builtIn as $fixer) {
+            $body[] = [
+                $fixer->getName(),
+                in_array($fixer, $configured) ? "<fg=green;>\xE2\x9C\x94</>" : "<fg=red;>\xE2\x9C\x96</>",
```

Comment:
answering your question,

we don't find comparing object instances with `==` as good practice.
change it to `in_array($fixer->getName(), $configuredFixerNames, true)`

Reply:
Yes, I don't see it as a good practice, too.

But in this particular case, a non strict comparison is the only way I know.

Comparing strictly don't work.

But using the strict comparison, the check fails as the two objects (the one in the configured fixers and the one in the fixers list) aren't the same object.

    // From fixers list
    PhpCsFixer\Fixer\Phpdoc\AlignMultilineCommentFixer {#592
        -tokenKinds: array:1 [
            0 => 378
        ]
      #configuration: array:1 [
        "comment_type" => "phpdocs_only"
      ]
      #whitespacesConfig: PhpCsFixer\WhitespacesFixerConfig {#35
        -indent: "    "
        -lineEnding: "\n"
      }
      -configurationDefinition: PhpCsFixer\FixerConfiguration\FixerConfigurationResolver {#593
        -options: array:1 [
            0 => PhpCsFixer\FixerConfiguration\FixerOption {#595
            -name: "comment_type"
            -description: "Whether to fix PHPDoc comments only (`phpdocs_only`), any multi-line comment whose lines all start with an asterisk (`phpdocs_like`) or any multi-line comment (`all_multiline`)."
            -default: "phpdocs_only"
            -isRequired: false
            -allowedTypes: null
            -allowedValues: array:3 [
                0 => "phpdocs_only"
              1 => "phpdocs_like"
              2 => "all_multiline"
            ]
            -normalizer: null
          }
        ]
        -registeredNames: array:1 [
            0 => "comment_type"
        ]
      }
    }

    // From configured fixers
    PhpCsFixer\Fixer\Phpdoc\AlignMultilineCommentFixer {#939
        -tokenKinds: array:1 [
            0 => 378
        ]
      #configuration: array:1 [
        "comment_type" => "phpdocs_only"
      ]
      #whitespacesConfig: PhpCsFixer\WhitespacesFixerConfig {#1030
        -indent: "    "
        -lineEnding: "\n"
      }
      -configurationDefinition: PhpCsFixer\FixerConfiguration\FixerConfigurationResolver {#940
        -options: array:1 [
            0 => PhpCsFixer\FixerConfiguration\FixerOption {#942
            -name: "comment_type"
            -description: "Whether to fix PHPDoc comments only (`phpdocs_only`), any multi-line comment whose lines all start with an asterisk (`phpdocs_like`) or any multi-line comment (`all_multiline`)."
            -default: "phpdocs_only"
            -isRequired: false
            -allowedTypes: null
            -allowedValues: array:3 [
                0 => "phpdocs_only"
              1 => "phpdocs_like"
              2 => "all_multiline"
            ]
            -normalizer: null
          }
        ]
        -registeredNames: array:1 [
            0 => "comment_type"
        ]
      }
    }

Any ideas about how to solve this?

Is worth the time to use a workaround?

Reply:
As stated in my comment:
```
change it to in_array($fixer->getName(), $configuredFixerNames, true)
```

so, use **name** instead of full **object**

Reply:
@Aerendir or use https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/2.5/src/RuleSet.php#L240

Reply:
Ok, done.